catalog/core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/criteria/temporal/TemporalEvaluationCriteriaImpl.java

```diff
@@ -25,21 +25,36 @@
 
     public TemporalEvaluationCriteriaImpl(Date end, Date start, Date input) {
         super();
-        this.end = end;
-        this.start = start;
-        this.input = input;
+        if(end != null) {
```

Comment:
Reformatted, thanks. Part of the challenge of this PR is getting into older code. It looks like it is possible (yet not sure if ideal) for end to be null. So for the time being I will do validate.notNull on start and input. Longer term the pub sub code should be refactored/reimplemented so I'm going to try not to do too much here on it.